core/platform/email/mailgun_email_services.py

```diff
@@ -72,3 +72,51 @@ def send_mail(
     requests.post(
         mailgun_domain_name, auth=('api', feconf.MAILGUN_API_KEY),
         data=data)
+
+
+def send_bulk_mail(
+        sender_email, recipient_emails, subject, plaintext_body, html_body):
+    """Sends an email using mailgun api.
+
+    In general this function should only be called from
+    email_manager._send_email().
+
+    Args:
+      - sender_email: str. the email address of the sender. This should be in
+          the form 'SENDER_NAME <SENDER_EMAIL_ADDRESS>'.
+      - recipient_emails: list. list of the email addresses of recipients.
+      - subject: str. The subject line of the email.
+      - plaintext_body: str. The plaintext body of the email.
+      - html_body: str. The HTML body of the email. Must fit in a datastore
+          entity.
+
+    Raises:
+      Exception: if the configuration in feconf.py forbids emails from being
+        sent.
+      Exception: if mailgun api key is not stored in feconf.MAILGUN_API_KEY.
+      Exception: if mailgun domain name is not stored in
+        feconf.MAILGUN_DOMAIN_NAME.
+      (and possibly other exceptions, due to mail.send_mail() failures)
+    """
+    if not feconf.MAILGUN_API_KEY:
+        raise Exception('Mailgun API key is not available.')
+
+    if not feconf.MAILGUN_DOMAIN_NAME:
+        raise Exception('Mailgun domain name is not set.')
+
+    mailgun_domain_name = (
+        'https://api.mailgun.net/v3/%s/messages' % feconf.MAILGUN_DOMAIN_NAME)
+
+    if not feconf.CAN_SEND_EMAILS:
+        raise Exception('This app cannot send emails to users.')
+
+    data = {
+        'from': sender_email,
+        'to': recipient_emails,
```

Comment:
Thanks, sounds good!

I was thinking a bit how to do the templating (in the long run). I think we'll need to break up our templating vars into two parts: variables that get interpolated in oppia, and variables that get passed to the mail service to interpolate. Also, we should change the templating syntax to follow the mailgun syntax, and implement a function in the GAE mail service that mimics the mailgun interpolation behavior before sending the email. That should keep it reasonably clean, I think.

All this is for another issue, though. Could you file it, so that we remember to do it?

Thanks!